### PR TITLE
Unique kerberos ccaches for parallel connections.

### PR DESCRIPTION
**Pull request by ggirol-rc**
The "MEMORY" ccache is shared in a process.
If a client uses it to make parallel connections,
the same ccache may be used for several clients with distinct credentials.
To prevent such sharing we create a unique, dedicated ccache when necessary with krb5_cc_new_unique.

To destroy the ccaches we created, the struct KRB_CREDENTIALS is extended to express the ccache ownership.

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: winpr/libwinpr/sspi/Kerberos/kerberos.c [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ctype.h>
+#include <krb5/krb5.h>
```
Comment (akallabeth) [inline winpr/libwinpr/sspi/Kerberos/kerberos.c:31]: This include needs to be guarded for builds without working kerberos (looking at you apple)

**Review by akallabeth — CHANGES_REQUESTED** [review]

File: winpr/libwinpr/sspi/Kerberos/kerberos.c (hunk 1 above)
Comment (ggirol-rc) [reply winpr/libwinpr/sspi/Kerberos/kerberos.c]: I removed the include altogether as it seems to be brought by the later `#include "krb5glue.h"`

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by akallabeth — APPROVED** [review]
LGTM

**Pull request merged**